sov-modules/sov-modules-impl/bank/tests/transfer_test.rs

```diff
@@ -0,0 +1,295 @@
+        assert!(error
+            .to_string()
+            .contains("Value not found for prefix: 0xc166b1b9c394ac408de38dd16fdba54edfcb3f7502f42ed59f296b93216f34f4 and: storage key"));
+
+        let receiver_balance_after = query_user_balance(receiver_address, &mut working_set);
+        assert_eq!(receiver_balance_before, receiver_balance_after);
+    }
+
+    // Receiver does not exist
+    {
+        let unknown_receiver = generate_address("non_existing_receiver");
+
+        let receiver_balance_before =
+            query_user_balance(unknown_receiver.clone(), &mut working_set);
+        assert!(receiver_balance_before.is_none());
+
+        let transfer_message = CallMessage::Transfer {
+            to: unknown_receiver.clone(),
+            coins: Coins {
+                amount: 1,
+                token_address: token_address.clone(),
+            },
+        };
+
+        let transferred = bank
+            .call(transfer_message, &sender_context, &mut working_set)
+            .expect("Transfer call failed");
+        assert!(transferred.events.is_empty());
+
+        let receiver_balance_after = query_user_balance(unknown_receiver, &mut working_set);
+        assert_eq!(Some(1), receiver_balance_after)
+    }
+
+    // Sender equals receiver
+    {
+        let total_supply_before = query_total_supply(&mut working_set);
+        let sender_balance_before = query_user_balance(sender_address.clone(), &mut working_set);
+        assert!(sender_balance_before.is_some());
+
+        let transfer_message = CallMessage::Transfer {
+            to: sender_address.clone(),
+            coins: Coins {
+                amount: 1,
+                token_address: token_address.clone(),
+            },
+        };
+        let transferred = bank
+            .call(transfer_message, &sender_context, &mut working_set)
+            .expect("Transfer call failed");
+        assert!(transferred.events.is_empty());
+
+        let sender_balance_after = query_user_balance(sender_address, &mut working_set);
+        assert_eq!(sender_balance_before, sender_balance_after);
+        let total_supply_after = query_total_supply(&mut working_set);
+        assert_eq!(total_supply_after, total_supply_before);
+    }
+}
+
+#[test]
+fn transfer_deployed_token() {
+    let bank = Bank::<C>::new();
+    let mut working_set = WorkingSet::new(ProverStorage::temporary());
+    let empty_bank_config = BankConfig::<C> { tokens: vec![] };
+    bank.genesis(&empty_bank_config, &mut working_set).unwrap();
+
+    let sender_address = generate_address("just_sender");
+    let receiver_address = generate_address("just_receiver");
+
+    let salt = 10;
+    let token_name = "Token1".to_owned();
+    let initial_balance = 1000;
+    let token_address = create_token_address::<C>(&token_name, sender_address.as_ref(), salt);
+
+    assert_ne!(sender_address, receiver_address);
+
+    // Preparation
+    let query_user_balance =
```

Comment:
Let's refactor `query_user_balance & query_total_supply` as helpers. It looks like they are used in several places.

Reply:
Although it does repeat few times, actual value of this function is to capture token address inside closure, so it can be skipped on follow calls. Extracting this into its own function won't bring that much benefit in my opinion.